youtube_dl/extractor/instagram.py

```diff
@@ -173,6 +188,27 @@ def _real_extract(self, url):
                 media = try_get(
                     additional_data, lambda x: x['graphql']['shortcode_media'],
                     dict)
+                if not media:
+                    # In some cases, the response from the server has items in it
+                    # Here is a sample of code that works fine with one item, but
+                    # I don't know for multiple items, cuz I don't have test links
+                    # for those.
+                    items0 = try_get(additional_data, lambda x: x['items'][0], dict)
+                    if items0:
+                        best_quality = items0.get("video_versions")[0]
```

Comment:
Have you checked if this works with the multi-video post test? IIRC you'd need to extract `carousel_media` for it

also
```suggestion
                        best_quality = items0['video_versions'][0]
```
or make the array indexing optional

Reply:
> Have you checked if this works with the multi-video post test? IIRC you'd need to extract `carousel_media` for it
> 
> also
> 
> or make the array indexing optional

Yeah I do check for multiple videos and I have proper extraction logic in [my repo](https://github.com/Udhayarajan/VidSnapKit/blob/ce8ba10a9e87d8e1af7626ffec3bbd6f4e7f9980/extractorlibrary/src/main/java/com/mugames/vidsnapkit/extractor/Instagram.kt#L235)